File: csharp/src/DarkStrata.CredentialCheck/Types/CheckOptions.cs
using System;

namespace DarkStrata.CredentialCheck;

/// <summary>
/// Options for individual credential check requests.
/// </summary>
public sealed class CheckOptions
{
    /// <summary>
    /// Client-provided HMAC key for deterministic results.
    /// </summary>
    /// <remarks>
    /// When provided, results are consistent across requests (not time-windowed).
    /// Must be a cryptographically strong hex string of at least 64 characters (256 bits).
    ///
    /// Use this when you need:
    /// - Consistent results across multiple requests
    /// - To avoid server-side HMAC key rotation
    /// - Custom key management
    /// </remarks>
    public string? ClientHmac { get; init; }

    /// <summary>
    /// Filter results to only include breaches from this date onwards.
    /// </summary>
    /// <remarks>
    /// Accepts either:
    /// - A DateTimeOffset which will be converted to epoch day
    /// - An integer epoch day (days since 1 January 1970)
    /// </remarks>
    public DateTimeOffset? Since { get; init; }

    /// <summary>
    /// Filter results to only include breaches from this epoch day onwards.
    /// </summary>
    /// <remarks>
    /// Epoch day = days since 1 January 1970 (e.g., 19724 = 1 January 2024).
    /// If both Since and SinceEpochDay are set, SinceEpochDay takes precedence.
    /// </remarks>
    public int? SinceEpochDay { get; init; }
}

File: credential-check/java/src/main/java/io/darkstrata/credentialcheck/CheckOptions.java
package io.darkstrata.credentialcheck;

import java.time.Instant;
import java.time.LocalDate;
import java.time.ZoneOffset;

/**
 * Options for individual check operations.
 */
public class CheckOptions {

    private final String clientHmac;
    private final Long since;

    private CheckOptions(Builder builder) {
        this.clientHmac = builder.clientHmac;
        this.since = builder.since;
    }

    /**
     * Get the client-provided HMAC key (at least 64 hex characters).
     */
    public String getClientHmac() {
        return clientHmac;
    }

    /**
     * Get the 'since' filter as epoch day (days since 1970-01-01).
     */
    public Long getSince() {
        return since;
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        private String clientHmac;
        private Long since;

        private Builder() {
        }

        /**
         * Set a client-provided HMAC key (at least 64 hex characters / 256 bits).
         */
        public Builder clientHmac(String clientHmac) {
            this.clientHmac = clientHmac;
            return this;
        }

        /**
         * Filter results to only include breaches since the given epoch day.
         */
        public Builder since(long epochDay) {
            this.since = epochDay;
            return this;
        }

        /**
         * Filter results to only include breaches since the given date.
         */
        public Builder since(LocalDate date) {
            this.since = date.toEpochDay();
            return this;
        }

        /**
         * Filter results to only include breaches since the given instant.
         */
        public Builder since(Instant instant) {
            this.since = instant.atZone(ZoneOffset.UTC).toLocalDate().toEpochDay();
            return this;
        }

        public CheckOptions build() {
            return new CheckOptions(this);
        }
    }
}
